enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,454 @@
+
+Since there can be a version skew between the kubelet and the kube-apiserver, and the feature gates are not assumed to be homogeneous, then it cannot be assumed that every
+kubelet in the cluster will be new enough to support UserNamespaces.
+
+While this would not be an issue if the featuregates are enabled on kubelet nodes after 4.15, we cannot assume the feature gate will be enabled.
+
+Thus, we need a method for the kube-apiserver to know for certain a node joining the cluster has a Kubelet that will also have the same featuregates set.
+A way to do this that also could set a precedent for Openshift to have the apiserver be more concious about kubelet version would be an openshift-apiserver
+extension that refuses to let a kubelet join a cluster or get leases if it is not new enough.
+
+For such a feature, there could be an extension to the NodeConfig object called `minimumKubeletVersion`. This field would be set on the cluster level. Practically, the
+openshift-apiserver could then query the `Node.Status.NodeSystemInfo.KubeletVersion` and verify it is above the `minimumKubeletVersion`.
+
+It is also possible this feature should be paired with a corresponding kubelet field `minimumKubeletVersion`, where it exits is it is too old. This will prevent the kubelet from
+running before it seeks to get credentials from the kube-apiserver, but also adds additional code overhead and backporting, plus given this feature would be added in z-streams, it's not
+possible to rely on.
+
+For this feature, there should be extensive documentation on what to do if this condition triggers. For instance, there could be situations where that would make a node completely
+unrecoverable, and we should ensure customers can reclaim their nodes and ensure they are new enough.
+
+1: I actually can't find any reference to this, but this is the way it *should* work, and thus we will rely on this.
+
+### Risks and Mitigations
+
+- Allowing user namespaces does open Openshift users to theoretical kernel vulnerabilities
+    - While user namespaces have existed for a while, kernel concepts like idmapped mounting
+      have not, and the newness could be seen as risky.
+    - However, the tangible security advantages of allowing user namespaces outweigh the theoretical
+      security risks.
+    - Plus, for the majority of users, user namespaces will not be enabled to begin with.
+
+### Drawbacks
+
+## Open Questions [optional]
+
+## Test Plan
+
+- e2e tests for a user namespaced pod, especially with different volume types to verify kernel idmapped mount support
+    - Also verify that the pod is actually in a user namespace
+- upgrade tests where some worker pools are paused and the apiserver has the feature enabled, to verify kubelets won't create `hostUsers: false` pods if not confined by a user namespace
+- long-term: e2e tests for running podman in a pod, so we have an established test path for users to know what works and what doesn't.
```

Comment:
this needs to be done prior to moving `scc/nested-container` out of techpreview

Reply:
added a note about this